torch/fx/experimental/proxy_tensor.py

```diff
@@ -96,9 +44,21 @@ def decompose(decomposition_table):
     finally:
         CURRENT_DECOMPOSITION_TABLE = old_decomposition_table
 
+def track_metadata(tensor, proxy, tracer):
+    for i, s in enumerate(tensor.shape):
+        if isinstance(s, SymInt):
+            inner_s = s.get_pyobj()
+            assert isinstance(inner_s, PySymInt)
+            # TODO: improve naming
+            # TODO: lazily insert this into the graph only on first
+            # use?  Maybe complicated and DCE is a better idea
+            inner_s.__dict__[tracer] = proxy.size(i)
```

Comment:
non obvious from the code why you are doing this ( at least on initial reading)

Reply:
will add comment on next ci round. But the general idea is we're saying that "if you use this symint to do compute, it should show up as 'x = tensor.size(i); ... something using x ...'